internal/handlers/security_group.go

```diff
@@ -0,0 +1,479 @@
+	))
+	defer span.End()
+	k, err := uuid.Parse(c.Param("id"))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, models.NewBadPathParameterError("id"))
+		return
+	}
+	var securityGroup models.SecurityGroup
+	result := api.db.WithContext(ctx).
+		First(&securityGroup, "id = ?", k)
+	if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+		c.Status(http.StatusNotFound)
+		return
+	}
+	c.JSON(http.StatusOK, securityGroup)
+}
+
+func (api *API) secGroupsEnabled(c *gin.Context) bool {
+	secGroupsEnabled, err := api.fflags.GetFlag("security-groups")
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, models.NewApiInternalError(err))
+		return false
+	}
+	allowForTests := c.GetString("nexodus.secGroupsEnabled")
+	if (!secGroupsEnabled && allowForTests != "true") || allowForTests == "false" {
+		c.JSON(http.StatusMethodNotAllowed, models.NewNotAllowedError("security-groups support is disabled"))
+		return false
+	}
+	return true
+}
+
+// CreateSecurityGroup handles adding a new SecurityGroup
+// @Summary      Add SecurityGroup
+// @Id  		 CreateSecurityGroup
+// @Tags         SecurityGroup
+// @Description  Adds a new Security Group
+// @Accepts		 json
+// @Produce      json
+// @Param        organization_id   path      string  true "Organization ID"
+// @Param        SecurityGroup   body   models.AddSecurityGroup  true "Add SecurityGroup"
+// @Success      201  {object}  models.SecurityGroup
+// @Failure      400  {object}  models.BaseError
+// @Failure		 401  {object}  models.BaseError
+// @Failure      409  {object}  models.ConflictsError
+// @Failure		 429  {object}  models.BaseError
+// @Failure      500  {object}  models.BaseError
+// @Router       /api/organizations/{organization_id}/security_groups [post]
+func (api *API) CreateSecurityGroup(c *gin.Context) {
+	ctx, span := tracer.Start(c.Request.Context(), "CreateSecurityGroup")
+	defer span.End()
+
+	if !api.secGroupsEnabled(c) {
+		return
+	}
+
+	var request models.AddSecurityGroup
+	// Call BindJSON to bind the received JSON
+	if err := c.BindJSON(&request); err != nil {
+		c.JSON(http.StatusBadRequest, models.NewBadPayloadError())
+		return
+	}
+
+	if request.OrganizationId == uuid.Nil {
+		c.JSON(http.StatusBadRequest, models.NewFieldNotPresentError("org_id"))
+		return
+	}
+
+	if request.GroupName == "" {
+		c.JSON(http.StatusBadRequest, models.NewFieldNotPresentError("group_name"))
+		return
+	}
+
+	var org models.Organization
+	if res := api.db.WithContext(ctx).
+		Scopes(api.OrganizationIsOwnedByCurrentUser(c)).
+		First(&org, "id = ?", request.OrganizationId); res.Error != nil {
+		c.JSON(http.StatusNotFound, models.NewNotFoundError("organization"))
+		return
+	}
+
+	var sg models.SecurityGroup
+	err := api.transaction(ctx, func(tx *gorm.DB) error {
+		sg = models.SecurityGroup{
+			GroupName:        request.GroupName,
+			OrganizationId:   request.OrganizationId,
+			InboundRules:     request.InboundRules,
+			OutboundRules:    request.OutboundRules,
```

Comment:
How about some validation on the rules so we can reject the request now instead of hitting errors later trying to apply bogus rules

Reply:
Was planning on this on a  follow up PR. It is fairly involved to go through all of the scenarios and I want to be thorough here.

Reply:
OK, for each of the follow-up items, can you file an issue and then link to it from the comment thread? Then I'll mark it as resolved.

Reply:
Done, https://github.com/nexodus-io/nexodus/issues/1113

I will also summarize them all at the bottom of the PR once Ive gone through everything. Ty!